messaging/src/main/java/org/axonframework/messaging/QueueMessageStream.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.Context;
+
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * MessageStream implementation that uses a Queue to make elements available to a consumer.
+ *
+ * @param <M> The type of Message managed by this stream
+ */
+public class QueueMessageStream<M extends Message<?>> implements MessageStream<M> {
```

Comment:
It doesn't seem that this implementation is used by the `MessageStream` `default` methods, nor the static factory methods in there. Is this just a "nice to have," or is this used in a branch downstream that you're working on?

Reply:
A part from the above comment, and my usual JavaDoc comments, I don't have any remarks against adding this implementation!

Reply:
Yes, the intent was to use it in another pull request. It also served as a good test if the test suite and base implementations were good enough.

Reply:
Awesome, thanks for clarifying!